internal/provider/resource_gitlab_group_project_file_template_test.go

```diff
@@ -0,0 +1,149 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+	"os"
+	"strconv"
+	"testing"
+)
+
+func TestAccGitlabGroupProjectFileTemplate_basic(t *testing.T) {
+	// Since we do some manual setup in this test, we need to handle the test skip first.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skip(fmt.Sprintf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc))
+	}
+	baseGroup, firstProject, secondProject := setupTestData(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckProjectFileTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(baseGroup.ID, firstProject.ID),
+				Check: resource.ComposeTestCheckFunc(
+					// Note - we can't use the testAccCheckGitlabGroupAttributes, because that checks the TF
+					// state attributes, and file project template explicitly doesn't exist there.
+					testAccCheckGitlabGroupFileTemplateValue(baseGroup, firstProject),
+					testAccCheckGitlabGroupFileTemplateStateValue(baseGroup, firstProject),
```

Comment:
I think it makes sense to use the off the shelve functions when ever possibl
```suggestion
					resource.TestCheckResourceAttr("gitlab_group_project_file_template.linking_template", "group_id", baseGroup.ID),
					resource.TestCheckResourceAttr("gitlab_group_project_file_template.linking_template", "file_template_project_id", firstProject.ID),
```